pkg/operation/botanist/controlplane.go

```diff
@@ -1356,6 +1355,43 @@ func (b *Botanist) DefaultKubeAPIServerService() component.DeployWaiter {
 	)
 }
 
+// SNIState returns true if kube-apiserver service exists
+// and it's type of LoadBalancer.
+func (b *Botanist) SNIPhase(ctx context.Context) component.Phase {
+	var (
+		svc        = &corev1.Service{}
+		sniEnabled = b.APIServerSNIEnabled()
+	)
+
+	if b.K8sSeedClient.DirectClient().Get(
+		ctx,
+		client.ObjectKey{Name: v1beta1constants.DeploymentNameKubeAPIServer, Namespace: b.Shoot.SeedNamespace},
+		svc,
+	) != nil {
+		if sniEnabled {
+			return component.PhaseEnabled
+		}
+
+		return component.PhaseDisabled
+	}
```

Comment:
Doesn't it make more sense to handle the error properly here instead of making some assumptions?